src/backend/dx12ll/src/command.rs

```diff
@@ -1,575 +1,726 @@
+            unsafe {
+                self.inner.CopyResource(
+                    dst.resource.as_mut_ptr(), // pDstResource
+                    src.resource.as_mut_ptr(), // pSrcResource
+                );
+            }
+        }
+    }
+
+    fn copy_image(&mut self, src: &native::Image, dest: &native::Image) {
+        unimplemented!()
+    }
+
+    fn copy_buffer_to_image(&mut self, buffer: &native::Buffer, image: &native::Image,
+                            _layout: memory::ImageLayout, regions: &[command::BufferImageCopy]) {
+        //TODO: handle layout?
+        let mut src = winapi::D3D12_TEXTURE_COPY_LOCATION {
+            pResource: buffer.resource.as_mut_ptr(),
+            Type: winapi::D3D12_TEXTURE_COPY_TYPE_PLACED_FOOTPRINT,
+            u: unsafe { mem::zeroed() },
+        };
+        let mut dest = winapi::D3D12_TEXTURE_COPY_LOCATION {
+            pResource: image.resource.as_mut_ptr(),
+            Type: winapi::D3D12_TEXTURE_COPY_TYPE_SUBRESOURCE_INDEX,
+            u: unsafe { mem::zeroed() },
+        };
+        let (width, height, depth, _) = image.kind.get_dimensions();
+        for region in regions {
+            assert_eq!(region.buffer_offset % winapi::D3D12_TEXTURE_DATA_PLACEMENT_ALIGNMENT as u64, 0);
+            assert_eq!(region.buffer_row_pitch % winapi::D3D12_TEXTURE_DATA_PITCH_ALIGNMENT as u32, 0);
+            assert!(region.buffer_row_pitch >= width as u32 * image.bits_per_texel as u32 / 8);
+            *unsafe { src.PlacedFootprint_mut() } = winapi::D3D12_PLACED_SUBRESOURCE_FOOTPRINT {
+                Offset: region.buffer_offset as UINT64,
+                Footprint: winapi::D3D12_SUBRESOURCE_FOOTPRINT {
+                    Format: image.dxgi_format,
+                    Width: width as UINT,
+                    Height: cmp::max(1, height as UINT),
+                    Depth: cmp::max(1, depth as UINT),
+                    RowPitch: region.buffer_row_pitch,
+                },
+            };
+            // TODO: `D3D12CalcSubresource`
+            *unsafe { dest.SubresourceIndex_mut() } = region.image_mip_level as UINT +
+                region.image_base_layer as UINT * image.kind.get_num_slices().unwrap_or(1) as UINT;
+            // fast path
+            unsafe {
+                self.inner.CopyTextureRegion(
+                    &dest,
+                    region.image_offset.x as UINT,
+                    region.image_offset.y as UINT,
+                    region.image_offset.z as UINT,
+                    &src, ptr::null(),
+                );
+            }
+        }
+    }
+
+    fn copy_image_to_buffer(&mut self) {
+        unimplemented!()
+    }
+
+    fn clear_color(&mut self, rtv: &native::RenderTargetView, value: command::ClearColor) {
+        let clear_color = match value {
+            command::ClearColor::Float(c) => c,
+            command::ClearColor::Int(c) => [c[0] as FLOAT, c[1] as FLOAT, c[2] as FLOAT, c[3] as FLOAT], // TODO: error?

```

Comment:
(Casts are probably wrong here, requiring to keep the same bits but that's something we rather need to test in the future..)